packages/react-native-executorch/common/rnexecutorch/models/ocr/RecognizerUtils.cpp

```diff
@@ -0,0 +1,190 @@
+#include "RecognizerUtils.h"
+
+namespace rnexecutorch::ocr {
+cv::Mat softmax(const cv::Mat &inputs) {
+  cv::Mat maxVal;
+  cv::reduce(inputs, maxVal, 1, cv::REDUCE_MAX, CV_32F);
+  cv::Mat expInputs;
+  cv::exp(inputs - cv::repeat(maxVal, 1, inputs.cols), expInputs);
+  cv::Mat sumExp;
+  cv::reduce(expInputs, sumExp, 1, cv::REDUCE_SUM, CV_32F);
+  cv::Mat softmaxOutput = expInputs / cv::repeat(sumExp, 1, inputs.cols);
+
+  return softmaxOutput;
+}
+
+std::vector<float> sumProbabilityRows(const cv::Mat &matrix) {
+  std::vector<float> sums;
+  sums.reserve(matrix.rows);
+  for (int i = 0; i < matrix.rows; ++i) {
+    sums.push_back(cv::sum(matrix.row(i))[0]);
+  }
+  return sums;
+}
+
+void divideMatrixByRows(cv::Mat &matrix, const std::vector<float> &rowSums) {
+  for (int i = 0; i < matrix.rows; ++i) {
+    matrix.row(i) /= rowSums[i];
+  }
+}
+
+std::pair<std::vector<float>, std::vector<int32_t>>
```

Comment:
Maybe name it with some meaningful like separate it into struct `maxValuesAndIndices` or similar.